src/datastore.jl

```diff
@@ -1,10 +1,31 @@
 ## datastore
 using Distributed
 
-struct DRef
+mutable struct DRef
     owner::Int
     id::Int
     size::UInt
+    function DRef(owner, id, size)
+        d = new(owner, id, size)
+        poolref(d)
+        finalizer(x->@async(poolunref(x)), d)
```

Comment:
Does `@async` use threads by default? Can we have `Threads.@spawn` in a finalizer? That would be better because the main thread can get busy without stopping this GC process.

Reply:
It does not use threads, and I'm sure `@spawn` is fine if `@async` is fine (neither of them should require a task switch). I'll make the change.

Reply:
I just realized that we don't need the `@spawn`/`@async` here since `poolunref` can instead do them for remote unrefs, so I'll move this logic there instead.